From 73283f1332e90fbb6fe0caa39a038cb16b83a54c Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Wed, 5 Dec 2012 22:42:25 +0100
Subject: [PATCH] Revert "Also check parent mimetypes in
 protocolForArchiveMimetype()."

This reverts commit 4f296cfbced2c4ad54beec8f500ed2d3fc04ee05.
It breaks opening odt and other zip-based files in external apps.
That's what I get for delegating the testing :-)
BUG: 311214
---
 kio/kio/kprotocolmanager.cpp |   19 +------------------
 1 files changed, 1 insertions(+), 18 deletions(-)

diff --git a/kio/kio/kprotocolmanager.cpp b/kio/kio/kprotocolmanager.cpp
index 3f1cadd..502a3e6 100644
--- a/kio/kio/kprotocolmanager.cpp
+++ b/kio/kio/kprotocolmanager.cpp
@@ -47,7 +47,6 @@
 #include <ksharedconfig.h>
 #include <kstandarddirs.h>
 #include <kurl.h>
-#include <kmimetype.h>
 #include <kmimetypetrader.h>
 #include <kprotocolinfofactory.h>
 
@@ -1190,23 +1189,7 @@ QString KProtocolManager::protocolForArchiveMimetype( const QString& mimeType )
             }
         }
     }
-    const QString prot = d->protocolForArchiveMimetypes.value(mimeType);
-    if (!prot.isEmpty())
-        return prot;
-
-    // Check parent mimetypes
-    KMimeType::Ptr mime = KMimeType::mimeType(mimeType);
-    if (mime) {
-        const QStringList parentMimeTypes = mime->allParentMimeTypes();
-        Q_FOREACH(const QString& parentMimeType, parentMimeTypes) {
-            const QString res = d->protocolForArchiveMimetypes.value(parentMimeType);
-            if (!res.isEmpty()) {
-                return res;
-            }
-        }
-    }
-
-    return QString();
+    return d->protocolForArchiveMimetypes.value(mimeType);
 }
 
 #undef PRIVATE_DATA
-- 
1.7.7
